### ISSUE TITLE
Spacing between values and units

### ISSUE THREAD

**Issue by Myriad-Dreamin** [johanvx/typst-undergradmath#17]
The problem is not what I care, but I read your README and I knew it from my knowledge. 

The Thin space (in Typst, it is represented by`\u{2009}`) can be used for space between numbers and SI units. To use it, we can define some functions for Typst similar to those available in [siunitx](https://www.ctan.org/pkg/siunitx).

+ Graphic designers are oftenly using "thin space" if the "space" is not good.
+ The default breaks between value and units is `\u{2009}` from [siunitx-v2.sty](https://github.com/josephwright/siunitx/blob/1b08e976141dea220bd774c282e2cfd5ce08fc7c/siunitx-v2.sty)
+ If there is a very long long long sentence, Typst will break a thin space, whill LaTeX will not. And, Typst follows the Unicode Standard.

This is still a unicode trick, but works in Typst...

Reference:
+ an awesome blog post: [Adventures in space: Spaces](https://www.fontshop.com/content/adventures-in-space_spaces)
+ [Whitespace character - Wikipedia](https://en.wikipedia.org/wiki/Whitespace_character)
  > [Thin space](https://en.wikipedia.org/wiki/Thin_space); one-fifth (sometimes one-sixth) of an em wide. Recommended for use as a [thousands separator](https://en.wikipedia.org/wiki/Thousands_separator) for measures made with [SI units](https://en.wikipedia.org/wiki/SI_units#SI_writing_style). Unlike U+2002 to U+2008, its width may get adjusted in typesetting.[[3]](https://en.wikipedia.org/wiki/Whitespace_character#cite_note-5) HTML/XML named entity: `&thinsp;`, `&ThinSpace;`, LaTeX: `\,` (the LaTeX thin space is a no-break space)
+ [The Unicode Standard 5.0, printed edition, Chapter 6, Space Characters](https://www.unicode.org/versions/Unicode5.0.0/ch06.pdf)
  > The exception is U+2009 thin space, which sometimes gets adjusted.

**Comment by johanvx** [comment]
Thanks for your knowledge sharing!

I was thinking about doing some trick with Unicode spaces, but soon realized that it would be too much for this project.

I'll leave this issue open until there's a non-tricky way to improve this. Hopefully this value-unit feature will be officially supported.

**Comment by Myriad-Dreamin** [comment]
World changes rapidly. There is already a package dedicated to typesetting unit text, [metro](https://github.com/fenjalien/metro). It is not quite mature but deserves to mention in the document. @johanvx @fenjalien

**Comment by johanvx** [comment]
> World changes rapidly. There is already a package dedicated to typesetting unit text, [metro](https://github.com/fenjalien/metro). It is not quite mature but deserves to mention in the document. @johanvx @fenjalien

Yeah, metro is great and so is [unify](https://github.com/ChHecker/unify).

I am a bit hesitant to use packages as they are still experimental.

**Comment by Myriad-Dreamin** [comment]
> I am a bit hesitant to use packages as they are still experimental.

Could I ask the consideration here? The `@preview` namespace for packages is reserved forever to the current experimental package resolution as far as I know, so you're not worried to get document broken in future.

**Comment by johanvx** [comment]
> Could I ask the consideration here?

Honestly, the only reason is that I don't like the `@preview` namespace 😆 . Apart from this, I do love to use packages in the document.
